### PR TITLE
Merge Login, Profile, Home, Stats, and WeekProgress; enforce MVVM; add Fake Repos + tests

### PR DESCRIPTION
**Pull request by khlilinfo**
### PR: Merge Login, Profile, Home, Stats, and WeekProgress; enforce MVVM; add Fake Repos + tests
#45 
**Summary**

This PR integrates the Login, Profile, Home, Stats, and WeekProgress/DailyObjective screens into a single flow.
I enforced MVVM across all screens, added Fake repositories with hard-coded data (to be swapped with Firebase later), and wrote tests for the repositories.

**Why**

Some screens didn’t respect MVVM (UI owning logic, direct data access). I reconstructed the screen logic to push business/state handling into ViewModels and repos. This makes the codebase testable, maintainable, and ready to switch to Firebase with minimal churn.

**What changed**

- **Screens integrated**: Login, Profile, Home, Stats, WeekProgress/DailyObjective wired into navigation.
- **MVVM refactors (where needed)**:
          - Moved logic from Composables to ViewModels.
          - Introduced UiState models and one-way data flow.
         - UI now observes state; no direct data source calls from Composables.
- **Repositories**:
         - Added interfaces: AuthRepository, ProfileRepository, HomeRepository, StatsRepository, WeekProgressRepository.  
         - Implemented Fake*Repo versions with deterministic data for local dev/CI.
          - DI binds interfaces → Fake implementations (easy swap to Firebase later).
- **Testing**:
         - Repository unit tests for all Fake repos (verify returned data and edge cases).
         - Stabilized UI tests (test tags + waits/scrolls where needed).

**How to test**

1. Build & run (Debug).
2. Call different Screens in `MainActivity.kt` and check if data is rendered
3. Run tests:
`./gradlew testDebugUnitTest connectedDebugAndroidTest`
Repo tests should pass with deterministic data.

**Follow-ups**

- Replace fakes with Firebase/Firestore implementations and switch DI binding.
- Expand ViewModel tests (loading/error/empty states) and Compose UI tests.

**Notes**
Omitting repos for not-yet-ready screens; they’ll be introduced in a follow-up PR once the screens and contracts are finalized.

This code was implemented with the help of AI

### CODE REVIEW

**Review by alaaae — APPROVED** [review]
Good job! I see you changed some tests, I assume it is for CI, thank you for that!
Overall good work

File: app/src/main/java/com/android/sample/feature/homeScreen/HomeRepository.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    return CreatureStats()
+  }
+
+  override suspend fun fetchUserStats(): UserStats {
+    return UserStats()
+  }
+
+  override fun dailyQuote(nowMillis: Long): String {
+    val idx = ((nowMillis / 86_400_000L) % quotes.size).toInt()
+    return quotes[idx]
```
Comment (DellaSantaF) [inline app/src/main/java/com/android/sample/feature/homeScreen/HomeRepository.kt:64]: Remember to write documentation for public methods.

File: app/src/main/java/com/android/sample/feature/homeScreen/HomeViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    val isLoading: Boolean = true,
+    val todos: List<ToDo> = emptyList(),
+    val creatureStats: CreatureStats = CreatureStats(),
+    val userStats: UserStats = UserStats(),
+    val quote: String = "",
+)
+
+// ---------- ViewModel ----------
+class HomeViewModel(
+    private val repository: HomeRepository = FakeHomeRepository(), // swap pour DI/Hilt
```
Comment (DellaSantaF) [inline app/src/main/java/com/android/sample/feature/homeScreen/HomeViewModel.kt:24]: Do not use a fake repository inside the class definition code, pass it as argument or swap for real repository.

File: app/src/main/java/com/android/sample/feature/weeks/repository/ObjectivesRepository.kt [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    val from = fromIndex.coerceIn(0, items.lastIndex)
+    val to = toIndex.coerceIn(0, items.lastIndex)
+    if (from == to) return items.toList()
+    val item = items.removeAt(from)
+    items.add(to, item)
+    return items.toList()
+  }
+
+  override suspend fun setObjectives(objs: List<Objective>): List<Objective> {
+    items.clear()
```
Comment (DellaSantaF) [inline app/src/main/java/com/android/sample/feature/weeks/repository/ObjectivesRepository.kt:76]: Once more, documentation on public methods.

File: app/src/main/java/com/android/sample/feature/weeks/viewmodel/ObjectivesViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import kotlinx.coroutines.launch
+
+// Holds only objective-related UI state
+data class ObjectivesUiState(
+    val objectives: List<Objective> = emptyList(),
+    val showWhy: Boolean = true,
+)
+
+class ObjectivesViewModel(
+    private val repository: ObjectivesRepository = FakeObjectivesRepository,
```
Comment (DellaSantaF) [inline app/src/main/java/com/android/sample/feature/weeks/viewmodel/ObjectivesViewModel.kt:21]: Same as before, avoid fake repository in constructor.

File: app/src/main/java/com/android/sample/feature/weeks/viewmodel/WeeksViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+data class WeeksUiState(
+    val weekProgressPercent: Int = 0,
+    val dayStatuses: List<DayStatus> = emptyList(),
+    val weeks: List<WeekProgressItem> = emptyList(),
+    val selectedWeekIndex: Int = 0,
+    val isLoading: Boolean = true,
+)
+
+class WeeksViewModel(
+    private val repository: WeeksRepository = FakeWeeksRepository(),
```
Comment (DellaSantaF) [inline app/src/main/java/com/android/sample/feature/weeks/viewmodel/WeeksViewModel.kt:26]: Same as before, avoid using fake repository in constructor.

File: app/src/main/java/com/android/sample/ui/profile/ProfileRepository.kt [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+/** Simple in-memory fake repository for the profile feature. */
+class FakeProfileRepository(initial: UserProfile = UserProfile()) : ProfileRepository {
+  private val _profile = MutableStateFlow(initial)
+  override val profile: StateFlow<UserProfile> = _profile
+
+  override suspend fun updateProfile(newProfile: UserProfile) {
+    _profile.value = newProfile
+  }
```
Comment (DellaSantaF) [inline app/src/main/java/com/android/sample/ui/profile/ProfileRepository.kt:20]: Remember to add documentation to public methods.

File: app/src/main/java/com/android/sample/ui/stats/repository/FakeStatsRepository.kt
Comment (DellaSantaF) [inline app/src/main/java/com/android/sample/ui/stats/repository/FakeStatsRepository.kt:1]: Avoid hard coded strings inside the code, use the strings.xml file.

**Review by DellaSantaF — APPROVED** [review]
Very well structured code, good use of repositories. I did some remarks on some repositories and viewModels that apply to all of them (Fake repo instanciation in constructor, documentation, hard coded strings), make sure to be careful to these notions in every file.
